server/core.c

```diff
@@ -5971,6 +5977,30 @@ static int core_upgrade_storage(request_rec *r)
     return DECLINED;
 }
 
+static apr_status_t core_get_conn_in_pollfd(conn_rec *c,
+                                         struct apr_pollfd_t *pfd,
+                                         apr_interval_time_t *ptimeout)
+{
+    if (c && !c->master) {
+        pfd->desc_type = APR_POLL_SOCKET;
+        pfd->desc.s = ap_get_conn_socket(c);
+        if (ptimeout) {
+            apr_socket_timeout_get(pfd->desc.s, ptimeout);
+        }
+        return APR_SUCCESS;
+    }
+    return DECLINED;
+}
+
+AP_CORE_DECLARE(apr_status_t) ap_get_conn_in_pollfd(conn_rec *c,
+                                      struct apr_pollfd_t *pfd,
+                                      apr_interval_time_t *ptimeout)
+{
+    apr_status_t rv;
+    rv = ap_run_get_conn_in_pollfd(c, pfd, ptimeout);
+    return (rv == DECLINED)? APR_ENOTIMPL : rv;
```

Comment:
If get_conn_in_pollfd hooks return APR_ENOTIMPL when they don't implement the thing we could use ap_run_get_conn_in_pollfd() directly possibly.

Reply:
Can do, yes. I find DECLINED better, since hooks commonly use that for calls where they do not engage. But that would work as well.

Reply:
Changed to APR_ENOTIMPL.